### PR TITLE
chore: update folder structure

### PR DESCRIPTION
**Pull request by anthony9187**
This PR updates the folder structure to break components out into `elements` and `layout` folders, as well as normalizing the mixins folder structure. 

This PR should not be merged until after [PR 49](https://github.com/rdkcentral/Lightning-UI-Components/pull/49) has been approved and merged.

### CODE REVIEW

**Review by THoj13 — APPROVED** [review]
It looks like this includes [your Row PR changes](https://github.com/rdkcentral/Lightning-UI-Components/pull/49/commits). Not sure if that will impact anything if it is merged after. Otherwise, all looks good to me.

**Comment by anthony9187** [comment]
> It looks like this includes [your Row PR changes](https://github.com/rdkcentral/Lightning-UI-Components/pull/49/commits). Not sure if that will impact anything if it is merged after. Otherwise, all looks good to me.

that's correct, this PR was branched off of `feat/update-focus-manager` in order to make the future merge a bit easier. That's why this PR needs to wait until the other one is merged. We could, of course, just merge this PR and include it all, but I'd rather have the git history/changelog reflect these as two separate changes

**Review by erautenberg — APPROVED** [review]

**Pull request merged**
